tests/Traits/Http/RequestMockHelpers.php

```diff
@@ -35,13 +35,14 @@ public function setRequestUserModel(User $user = null): void
     /**
      * Generates a new request user model and also returns the generated model.
      */
-    public function generateRequestUserModel(array $args = []): User
+    public function generateRequestUserModel(bool $isRootAdmin, array $args = []): void
     {
-        /** @var \App\Models\User $user */
         $user = User::factory()->make($args);
-        $this->setRequestUserModel($user);
+        $user = m::mock($user)->makePartial();
+        $user->shouldReceive('isRootAdmin')->andReturn($isRootAdmin);
```

Comment:
I don't like mocking and having the assertion this every single time this is called, I'd rather just generate the permission/role for real here.

Reply:
I had to do mocking because we run tests without migrations.

Reply:
Ah, forgot about that, then this is fine, thanks